app/models/cart_item/validator.rb

```diff
@@ -10,19 +10,23 @@ def initialize(cart_item)
 
         def validate!
           raise CartItem::InvalidProductIdError unless product_id_valid?
-          raise CartItem::InvalidQuantityError unless quantity_valid?
+          raise CartItem::InvalidQuantityTypeError unless quantity_type_valid?
+          raise CartItem::InvalidQuantityValueError unless quantity_value_valid?
         end
 
         private
 
         def product_id_valid?
-          product = @cart_item.product
-          !product.nil?
+          !@cart_item.product.nil?
```

Comment:
You really dislike the `nil?` method. :) I'm using it as it seems more readable, but will try to remember to just use the truthy nature of non-`nil` values in the future.